### DISCUSSION TITLE
how add custom attributes to my own Setting Model

### DISCUSSION THREAD

**Discussion by muhamedRadwan** [spatie/laravel-settings#79] [Q&A]
How I can make custom attributes in Setting Model 
![image](https://user-images.githubusercontent.com/16479089/126099041-af78f6d0-9696-4c91-8202-23b7aeecfe1c.png)

**Comment by muhamedRadwan** [comment]
why use Spatie\LaravelSettings\Settings
not extends  Illuminate\Database\Eloquent\Model  ?

**Reply by muhamedRadwan** [reply]
I made changes on plugin and made it extends from  Illuminate\Database\Eloquent\Model

![image](https://user-images.githubusercontent.com/16479089/126117307-035e69d3-7f52-4431-9e52-8ad5c3425330.png)
and made also changes to  SettingsConfig
![image](https://user-images.githubusercontent.com/16479089/126117371-245e0b2a-8bd0-4235-bf5d-f02402e11a9f.png)
and now working great with GetMutator

**Reply by muhamedRadwan** [reply]
but I want to know if this break anything because I think you have already not extended Eloquent\Model  for good reason

**Reply by rubenvanassche** [reply]
Hi @muhamedRadwan,

The settings class does not extend from `Eloquent\Model` since its built to be used with multiple repositories like Redis or your own implementation. If you're using the database repository, then take a look at the `Models/Settingsproperty` class this is an Eloquent model that can be extended.

**Reply by muhamedRadwan** [reply]
yes, I use the database repository but I can not use GetMutator ?

**Reply by muhamedRadwan** [reply]
also, do I can use spaite translate in this plugin ?

**Reply by rubenvanassche** [reply]
Hi @muhamedRadwan,

Though the package uses an eloquent model as the basis for the database repository, it just could use raw sql statements, the core of the package is the Settings class. And the model should just follow, so using things like spatie/translatable and mutator* methods isn't possible.

**Reply by muhamedRadwan** [reply]
I made a new Abstract Class Called SettingsEloquent that extends the Eloquent Model
and change in the base to workaround events 
and now can a user select to work with normal Settings class or SettingsEloquent  class
when he/she create his own Model 

also for not repeating the code, I made a trait that has all functions and variables that used in Settings Class 

and used it in settings and SettingsEloquent  

and in SettingsEloquent  I overriding the __get function to use getAttribute function from Illuminate\Database\Eloquent\Model

to be allowed to use getMutator and normal eloquent functions

and check all unit testing it passed 

also, make a new test for check getMutator  and it worked normally 

can I make a pull request to check together if you have time because it first time I contribute an open-source project and I'm really exciting to give community some help

**Reply by marawanaziz** [reply]
@rubenvanassche any update ?

**Reply by rubenvanassche** [reply]
Hi @marawanaziz ,

As explained in the PR @muhamedRadwan did, I don't think this package is the correct place to implement such functionality. I've listed some alternatives here:

https://github.com/spatie/laravel-settings/pull/81#issuecomment-901668000
